diff --git a/debian/control b/debian/control
index 409dc22..9f50630 100644
--- a/debian/control
+++ b/debian/control
@@ -17,6 +17,8 @@ Build-Depends: debhelper (>= 10),
                libperl-dev,
                libssl-dev,
                libxslt1-dev,
+               libcurl4-openssl-dev,
+               ruby-dev,
                po-debconf,
                quilt,
                zlib1g-dev
@@ -313,6 +315,16 @@ Description: Bring echo and more shell style goodies to Nginx
  3. Merge contents of various "Nginx locations" (i.e., subrequests) together in
     a single main request (using echo_location and its friends).
 
+Package: libnginx-mod-http-passenger
+Architecture: any
+Depends: ${misc:Depends}, ${shlibs:Depends}
+Description: Nginx integration for Passenger application server
+ Passenger is an application server for Ruby, Node.js, Python, Meteor.
+ It makes deployment of web apps very easy and robust. Passenger is a
+ Phusion product.
+ .
+ This package contains the Nginx module for integration with Passenger.
+
 Package: libnginx-mod-http-upstream-fair
 Architecture: any
 Depends: ${misc:Depends}, ${shlibs:Depends}
diff --git a/debian/libnginx-mod-http-passenger.nginx b/debian/libnginx-mod-http-passenger.nginx
new file mode 100755
index 0000000..78c206f
--- /dev/null
+++ b/debian/libnginx-mod-http-passenger.nginx
@@ -0,0 +1,13 @@
+#!/usr/bin/perl -w
+
+use File::Basename;
+
+# Guess module name
+$module = basename($0, '.nginx');
+$module =~ s/^libnginx-mod-//;
+
+$modulepath = $module;
+$modulepath =~ s/-/_/g;
+
+print "mod debian/build-extras/objs/ngx_${modulepath}_module.so\n";
+print "mod debian/libnginx-mod.conf/mod-${module}.conf\n";
diff --git a/debian/libnginx-mod.conf/mod-http-passenger.conf b/debian/libnginx-mod.conf/mod-http-passenger.conf
new file mode 100755
index 0000000..bba8811
--- /dev/null
+++ b/debian/libnginx-mod.conf/mod-http-passenger.conf
@@ -0,0 +1 @@
+load_module modules/ngx_http_passenger_module.so;
diff --git a/debian/modules/control b/debian/modules/control
index a5d65a8..c555615 100644
--- a/debian/modules/control
+++ b/debian/modules/control
@@ -16,6 +16,10 @@ Homepage: https://github.com/agentzh/echo-nginx-module
 Version: v0.61
 Files-Excluded: .gitignore .gitattributes .travis.yml
 
+Module: http-passenger
+Homepage: https://github.com/phusion/passenger
+Version: 6.0.4
+
 Module: http-lua
 Homepage: https://github.com/openresty/lua-nginx-module
 Version: 0.10.13
diff --git a/debian/modules/http-passenger/config b/debian/modules/http-passenger/config
new file mode 100755
index 0000000..0b8b262
--- /dev/null
+++ b/debian/modules/http-passenger/config
@@ -0,0 +1,20 @@
+PASSENGER_SOURCE_PATH=/build/passenger
+
+# We include from here the module in the passenger repository, this avoids changing
+# the structure of nginx build process.
+if test "x$PASSENGER_SOURCE_PATH" = "x"; then
+    echo "error: Please export variable PASSENGER_SOURCE_PATH and point it towards passenger source directory; aborting"
+    exit 1
+fi
+
+# Change the addon directory and continue from there
+ngx_addon_dir="$PASSENGER_SOURCE_PATH/src/nginx_module"
+
+# Check if directory exists
+if [ ! -d "$ngx_addon_dir" ]; then
+    echo "error: PASSENGER_SOURCE_PATH does not contain src/nginx_module; aborting"
+    exit 1
+fi
+
+# Continue
+. $ngx_addon_dir/config
diff --git a/debian/rules b/debian/rules
index 981a57e..4e65bbd 100755
--- a/debian/rules
+++ b/debian/rules
@@ -11,6 +11,7 @@ DYN_MODS := \
 	http-cache-purge \
 	http-dav-ext \
 	http-echo \
+	http-passenger \
 	http-fancyindex \
 	http-geoip \
 	http-headers-more-filter \
@@ -129,6 +130,7 @@ extras_configure_flags := \
 			--add-dynamic-module=$(MODULESDIR)/http-dav-ext \
 			--add-dynamic-module=$(MODULESDIR)/http-ndk \
 			--add-dynamic-module=$(MODULESDIR)/http-echo \
+			--add-dynamic-module=$(MODULESDIR)/http-passenger \
 			--add-dynamic-module=$(MODULESDIR)/http-fancyindex \
 			--add-dynamic-module=$(MODULESDIR)/nchan \
 			--add-dynamic-module=$(MODULESDIR)/http-lua \
diff --git a/debian/tests/control b/debian/tests/control
index b491482..7d2403b 100644
--- a/debian/tests/control
+++ b/debian/tests/control
@@ -26,6 +26,7 @@ Depends: nginx-full,
          libnginx-mod-http-cache-purge,
          libnginx-mod-http-dav-ext,
          libnginx-mod-http-echo,
+         libnginx-mod-http-passenger,
          libnginx-mod-http-fancyindex,
          libnginx-mod-http-geoip,
          libnginx-mod-http-headers-more-filter,
@@ -49,6 +50,7 @@ Depends: nginx-light,
          libnginx-mod-http-cache-purge,
          libnginx-mod-http-dav-ext,
          libnginx-mod-http-echo,
+         libnginx-mod-http-passenger,
          libnginx-mod-http-fancyindex,
          libnginx-mod-http-geoip,
          libnginx-mod-http-headers-more-filter,
@@ -72,6 +74,7 @@ Depends: nginx-extras,
          libnginx-mod-http-cache-purge,
          libnginx-mod-http-dav-ext,
          libnginx-mod-http-echo,
+         libnginx-mod-http-passenger,
          libnginx-mod-http-fancyindex,
          libnginx-mod-http-geoip,
          libnginx-mod-http-headers-more-filter,
